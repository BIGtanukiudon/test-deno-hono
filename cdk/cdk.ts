import { App, Duration, Stack, StackProps } from 'aws-cdk-lib'
import lambda from 'aws-cdk-lib/aws-lambda'
import ecr from 'aws-cdk-lib/aws-ecr'
import iam from 'aws-cdk-lib/aws-iam'
import { HttpLambdaIntegration } from 'aws-cdk-lib/aws-apigatewayv2-integrations'
import apiGateway from 'aws-cdk-lib/aws-apigatewayv2'

export class AppStack extends Stack {
  constructor(scope: App, id: string, props?: StackProps) {
    super(scope, id, props)

    const ecrRepoArn = ecr.Repository.fromRepositoryArn(
      this,
      'test-deno-hono',
      'arn:aws:ecr:ap-northeast-1:123456789000:repository/test-deno-hono',
    )

    const iamRole = new iam.Role(this, 'TestDenoHonoRole', {
      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName(
          'service-role/AWSLambdaBasicExecutionRole',
        ),
      ],
    })

    const testDenoHonoLambda = new lambda.DockerImageFunction(
      this,
      'TestDenoHono',
      {
        code: lambda.DockerImageCode.fromEcr(ecrRepoArn),
        memorySize: 512,
        timeout: Duration.seconds(180),
        role: iamRole,
      },
    )

    const integration = new HttpLambdaIntegration(
      'TestDenoHonoApiIntegration',
      testDenoHonoLambda,
    )

    new apiGateway.HttpApi(this, 'TestDenoHonoApi', {
      defaultIntegration: integration,
      corsPreflight: {
        allowOrigins: ['*'],
        allowMethods: [
          apiGateway.CorsHttpMethod.GET,
          apiGateway.CorsHttpMethod.POST,
          apiGateway.CorsHttpMethod.OPTIONS,
        ],
        exposeHeaders: ['*'],
      },
    })
  }
}

new AppStack(new App(), 'test-deno-hono-lambda-stack', {
  env: { region: 'ap-northeast-1', account: '123456789000' },
})
