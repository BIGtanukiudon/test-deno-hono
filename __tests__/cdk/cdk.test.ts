import { assertSnapshot } from '@std/testing/snapshot'
import { Template } from 'aws-cdk-lib/assertions'
import { App } from 'aws-cdk-lib'
import { AppStack } from '$/cdk/cdk.ts'

Deno.test('Snapshot Test', async (t) => {
  const stack = new AppStack(new App(), 'test-deno-hono-lambda-stack')
  await assertSnapshot(t, Template.fromStack(stack).toJSON())
})
