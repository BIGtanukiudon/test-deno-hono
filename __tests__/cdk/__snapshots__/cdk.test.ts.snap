export const snapshot = {};

snapshot[`Snapshot Test 1`] = `
{
  Parameters: {
    BootstrapVersion: {
      Default: "/cdk-bootstrap/hnb659fds/version",
      Description: "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      Type: "AWS::SSM::Parameter::Value<String>",
    },
  },
  Resources: {
    TestDenoHono495DCBBB: {
      DependsOn: [
        "TestDenoHonoRole5F2C8083",
      ],
      Properties: {
        Code: {
          ImageUri: {
            "Fn::Join": [
              "",
              [
                "123456789000.dkr.ecr.ap-northeast-1.",
                {
                  Ref: "AWS::URLSuffix",
                },
                "/test-deno-hono:latest",
              ],
            ],
          },
        },
        MemorySize: 512,
        PackageType: "Image",
        Role: {
          "Fn::GetAtt": [
            "TestDenoHonoRole5F2C8083",
            "Arn",
          ],
        },
        Timeout: 180,
      },
      Type: "AWS::Lambda::Function",
    },
    TestDenoHonoApiDefaultStage7F84252F: {
      Properties: {
        ApiId: {
          Ref: "TestDenoHonoApiF300FCC6",
        },
        AutoDeploy: true,
        StageName: "\$default",
      },
      Type: "AWS::ApiGatewayV2::Stage",
    },
    TestDenoHonoApiF300FCC6: {
      Properties: {
        Name: "TestDenoHonoApi",
        ProtocolType: "HTTP",
      },
      Type: "AWS::ApiGatewayV2::Api",
    },
    TestDenoHonoApiGET72A56996: {
      Properties: {
        ApiId: {
          Ref: "TestDenoHonoApiF300FCC6",
        },
        AuthorizationType: "NONE",
        RouteKey: "GET /",
        Target: {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                Ref: "TestDenoHonoApiGETTestDenoHonoApiIntegration556B8026",
              },
            ],
          ],
        },
      },
      Type: "AWS::ApiGatewayV2::Route",
    },
    TestDenoHonoApiGETTestDenoHonoApiIntegration556B8026: {
      Properties: {
        ApiId: {
          Ref: "TestDenoHonoApiF300FCC6",
        },
        IntegrationType: "AWS_PROXY",
        IntegrationUri: {
          "Fn::GetAtt": [
            "TestDenoHono495DCBBB",
            "Arn",
          ],
        },
        PayloadFormatVersion: "2.0",
      },
      Type: "AWS::ApiGatewayV2::Integration",
    },
    TestDenoHonoApiGETTestDenoHonoApiIntegrationPermission9674020C: {
      Properties: {
        Action: "lambda:InvokeFunction",
        FunctionName: {
          "Fn::GetAtt": [
            "TestDenoHono495DCBBB",
            "Arn",
          ],
        },
        Principal: "apigateway.amazonaws.com",
        SourceArn: {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                Ref: "AWS::Partition",
              },
              ":execute-api:",
              {
                Ref: "AWS::Region",
              },
              ":",
              {
                Ref: "AWS::AccountId",
              },
              ":",
              {
                Ref: "TestDenoHonoApiF300FCC6",
              },
              "/*/*/",
            ],
          ],
        },
      },
      Type: "AWS::Lambda::Permission",
    },
    TestDenoHonoRole5F2C8083: {
      Properties: {
        AssumeRolePolicyDocument: {
          Statement: [
            {
              Action: "sts:AssumeRole",
              Effect: "Allow",
              Principal: {
                Service: "lambda.amazonaws.com",
              },
            },
          ],
          Version: "2012-10-17",
        },
        ManagedPolicyArns: [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  Ref: "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      Type: "AWS::IAM::Role",
    },
  },
  Rules: {
    CheckBootstrapVersion: {
      Assertions: [
        {
          Assert: {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    Ref: "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          AssertDescription: "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
